generators/upgrade/index.js

```diff
@@ -0,0 +1,180 @@
+    semver = require('semver'),
+    Git = require('git-wrapper');
+
+var UpgradeGenerator = generators.Base.extend({});
+
+util.inherits(UpgradeGenerator, scriptBase);
+
+/* Constants used throughout */
+const GENERATOR_JHIPSTER = 'generator-jhipster',
+    UPGRADE_BRANCH = 'jhipster_upgrade';
+
+module.exports = UpgradeGenerator.extend({
+    constructor: function () {
+        generators.Base.apply(this, arguments);
+
+        this.currentVersion = packagejs.version;
+        this.force = true;
+        this.git = new Git();
+
+        this.gitCheckout = function(branch) {
+            var done = this.async();
+            this.git.exec('checkout', {q: true}, [branch], function(err, msg) {
+                if (err != null) throw new Error(err);
+                this.log('Checked out branch ' + branch);
+                done();
+            }.bind(this));
+        };
+    },
+
+    initializing: {
+        displayLogo: function () {
+            this.printJHipsterLogo();
+        }
+    },
+
+    configuring: {
+        assertGitPresent: function() {
+            var done = this.async();
+            exec('git --version', function (err) {
+                if (err) {
+                    this.log(chalk.yellow.bold('WARNING!') + ' git is not found on your computer.\n',
+                        ' Install git: ' + chalk.yellow('http://git-scm.com/')
+                    );
+                    return;
+                }
+                done();
+            }.bind(this));
+        },
+
+        assertGitRepository: function() {
+            if (! fs.existsSync('.git')) {
+                this.log(chalk.yellow.bold('WARNING!') + ' no git repository found.\n',
+                    ' Please initialize a git repository here');
+                return;
+            }
+        },
+
+        assertNoLocalChanges: function() {
+            var done = this.async();
+            this.git.exec('status', {}, ['--porcelain'], function(err, msg) {
+                if (err != null) throw new Error(err);
+                if (msg != null && msg !== '') {
+                    this.log(chalk.yellow.bold('WARNING!') + ' local changes found.\n',
+                        ' Please commit/stash them before upgrading');
+                    return;
+                }
+                done();
+            }.bind(this));
+        },
+
+        checkLatestVersion: function() {
+            this.log('Looking for latest ' + GENERATOR_JHIPSTER + ' version...');
+            var done = this.async();
+            shelljs.exec('npm show ' + GENERATOR_JHIPSTER + ' version', {silent:true}, function (code, stdout, stderr) {
+                this.latestVersion = stdout.replace('\n','');
+                if (this.force || semver.lt(this.currentVersion, this.latestVersion)) {
+                    this.log(chalk.green('New ' + GENERATOR_JHIPSTER + ' version found: ' + this.latestVersion));
+                } else {
+                    this.log(chalk.green('No update available'));
```

Comment:
Use `this.error` method. Also the message should say that we are existing as the application is already generated using latest jhipster